solidity/test/liquidation-tests/liquidation-test-utils/createInstance.js

```diff
@@ -0,0 +1,229 @@
+}
+
+// move from AWAITING_SIGNER_SETUP to AWAITING_BTC_FUNDING_PROOF
+const toAwaitingFundingProof = async (testDeposit) => {
+    // retrieve pubkey and move to AWAITING_BTC_FUNDING_PROOF.
+    await testDeposit.retrieveSignerPubkey()
+}
+
+// move from AWAITING_BTC_FUNDING_PROOF to ACTIVE 
+const toActive = async (testDeposit) => {
+  // provide the funding proof and move to ACTIVE state.
+  await testDeposit.provideBTCFundingProof(
+    _version,
+    _txInputVector,
+    _txOutputVector,
+    _txLocktime,
+    _fundingOutputIndex,
+    _merkleProof,
+    _txIndexInBlock,
+    _bitcoinHeaders,
+  )
+}
+
+// move from ACTIVE to AWAITING_WITHDRAWAL_SIGNATURE
+const toAwaitingWithdrawalSignature = async (testDeposit, values = defaultValues) => {
+  await tbtcToken.resetBalance(signerFee, {from: values.depositOwner})
+
+  await tbtcToken.approve(testDeposit.address, signerFee, {
+    from: values.depositOwner,
+  })
+
+  await testDeposit.requestRedemption(
+    "0x1111111100000000",
+    redeemerOutputScript,
+    { from: values.depositOwner },// only TDT owner can redeem pre-term
+  )
+}
+
+// move from AWAITING_WITHDRAWAL_SIGNATURE to AWAITING_WITHDRAWAL_PROOF
+const toAwaitingWithdrawalProof = async (testDeposit, values = defaultValues) => {
+  await testDeposit.setSigningGroupPublicKey(pubkeyX, pubkeyY)
+  await testDeposit.setRequestInfo(
+    values.depositOwner,
+    "0x" + "11".repeat(20),
+    0,
+    0,
+    digest,
+  )
+  await testDeposit.provideRedemptionSignature(v, r, s)
+  
+}
+
+// Determine what the requires state is and call the appropriate functions to reach it.
+const determineStateTransitionPath = async (testDeposit, values = defaultValues) => {
+  switch(values.state){
+    case states.AWAITING_SIGNER_SETUP.toNumber():
+      break;
+    case states.AWAITING_BTC_FUNDING_PROOF.toNumber():
+      await toAwaitingFundingProof(testDeposit, values)
+      break;
+    case states.ACTIVE.toNumber():
+      await toAwaitingFundingProof(testDeposit, values)
+      await toActive(testDeposit, values)
+      break;
+    case states.AWAITING_WITHDRAWAL_SIGNATURE.toNumber():
+      await toAwaitingFundingProof(testDeposit, values)
+      await toActive(testDeposit, values)
+      await toAwaitingWithdrawalSignature(testDeposit, values)
+      break;
+    case states.AWAITING_WITHDRAWAL_PROOF.toNumber():
+      await toAwaitingFundingProof(testDeposit, values)
+      await toActive(testDeposit, values)
+      await toAwaitingWithdrawalSignature(testDeposit, values)
+      await toAwaitingWithdrawalProof(testDeposit, values)
+      break;
+    default:
+      throw new Error('required state destination does not have a supported path')
+  }
+}
+
+module.exports.createInstance = createInstance
+module.exports.toActive = toActive
+module.exports.toAwaitingFundingProof = toAwaitingFundingProof
+module.exports.toAwaitingWithdrawalSignature = toAwaitingWithdrawalSignature
+module.exports.toAwaitingWithdrawalProof = toAwaitingWithdrawalProof
```

Comment:
To what extent can we replace/have we replaced this with the state tests? It feels like it's doing very similar exploration of the state space.